### PR TITLE
linux-capture: Prevent X11 display/window captures from capturing implicitly

### PR DESCRIPTION
**Pull request by Lain-B**
### Description
There are two situations where xcomposite window capture will capture random windows: on first creation, and when going to the properties when the current window is invalid. The first happens because it apparently captures the first top-level window if there is no set value. The second happens because when opening the properties, the properties widget cannot find the value it's looking for and defaults to selecting the first one when the properties are opened, thus selecting and capturing the first window in the list (which is probably something we should fix in the properties view at some point but I don't want to dive into code that's even *more* cursed than xcomposite code right now).

So instead, this diff makes it so that on first creation, it creates a value that says "[Select a window to capture]" that keeps the capture inactive until a user actually chooses a window rather than the top-level window. It also makes it so that if the user has a window that is no longer valid, it will keep that window in the list and as the currently selected value, which prevents it from automatically selecting the first window in the list when properties are opened.

The xshm Display Capture has also been fixed similarly to not automatically capture a display on first creation.

### Motivation and Context
A window/display capture should not capture something without the user explicitly choosing it.

### How Has This Been Tested?
Tested it locally, might be good if others test it to ensure it works as intended and doesn't have any issues. 99.9% sure that it works fine though.

### Types of changes
- Tweak (non-breaking change to improve existing functionality)
- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.

### CODE REVIEW

File: plugins/linux-capture/xcomposite-input.c [hunk 1]
```diff
@@ -302,19 +294,45 @@ xcb_window_t xcomp_find_window(xcb_connection_t *conn, Display *disp,
...
+xcb_window_t xcomp_find_window(xcb_connection_t *conn, Display *disp,
+			       const char *str)
+{
+	xcb_window_t ret = 0;
+	char *wname = NULL;
+	char *wcls = NULL;
+	DARRAY(xcb_window_t) tlw = {0};
+
+	if (!str || strlen(str) == 0) {
+		if (tlw.num > 0)
```
Comment (kkartaltepe) [inline plugins/linux-capture/xcomposite-input.c]: (probably fixed in later commit but,) tlw is being referenced uninitialized, the line 333 initialization is supposed to occur before this check.

Comment (LainAmongYou) [reply plugins/linux-capture/xcomposite-input.c]: Fixed.

Comment (Lain-B) [reply plugins/linux-capture/xcomposite-input.c]: Fixed!

**Comment by Lain-B** [comment]
I also added a fix for xshm Display Capture as well.

**Pull request merged**